### PR TITLE
IGNITE-16719.

### PR DESCRIPTION
**Pull request by sanpwc**

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java [hunk 1]
```diff
@@ -542,6 +542,8 @@ public void subscribe(Subscriber<? super BinaryRow> subscriber) {
 
             private AtomicInteger scanCounter = new AtomicInteger(1);
 
+            private static final int INTERNAL_BATCH_SIZE = 10_000;
```
Comment (tledkov-gridgain) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java:545]: Should we use configured parameter to define internal batch size?

Comment (tledkov-gridgain) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java:545]: Should we use configured parameter to define internal batch size?

File: modules/table/src/integrationTest/java/org/apache/ignite/distributed/ItInternalTableScanTest.java [hunk 2]
```diff
@@ -527,7 +543,7 @@ public void onSubscribe(Subscription subscription) {
             public void onNext(BinaryRow item) {
                 retrievedItems.add(item);
 
-                if (retrievedItems.size() % reqAmount == 0) {
+                if (reqAmount != Long.MAX_VALUE && retrievedItems.size() % reqAmount == 0) {
```
Comment (tledkov-gridgain) [inline modules/table/src/integrationTest/java/org/apache/ignite/distributed/ItInternalTableScanTest.java:546]: What for additional  condition `reqAmount != Long.MAX_VALUE` is added?
Is the condition `retrievedItems.size() % reqAmount` **true** in this case?

Comment (tledkov-gridgain) [inline modules/table/src/integrationTest/java/org/apache/ignite/distributed/ItInternalTableScanTest.java:546]: What for additional  condition `reqAmount != Long.MAX_VALUE` is added?
Is the condition `retrievedItems.size() % reqAmount` **true** in this case?

**Review by tledkov-gridgain — APPROVED** [review]
OK with me

File: modules/table/src/integrationTest/java/org/apache/ignite/distributed/ItInternalTableScanTest.java (hunk 2 above)
Comment (sanpwc) [reply modules/table/src/integrationTest/java/org/apache/ignite/distributed/ItInternalTableScanTest.java:546]: Yep, it's possible to remove given check here. Fixed.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java [hunk 3]
```diff
@@ -607,7 +617,7 @@ private void cancel(boolean closeCursor) {
             /**
              * Requests and processes n requested elements where n is an integer.
              *
-             * @param n Requested amount of items.
+             * @param n amount of items to request and process.
```
Comment (korlov42) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java:620]:
```suggestion
             * @param n Amount of items to request and process.
```

**Review by korlov42 — APPROVED** [review]

**Pull request merged**